### PR TITLE
Update exemplar docs based on changes to exemplar storage configuration

### PR DESCRIPTION
**Pull request by cstyan**
@LeviHarrison pointed out that we hadn't updated the docs based on the changes from #8974

@roidelapluie not sure how we go about backporting docs into the right releases. Releases from 2.29 forward should have the docs changes from this PR (https://github.com/prometheus/prometheus/releases/tag/v2.29.0)

Signed-off-by: Callum Styan <callumstyan@gmail.com>

### CODE REVIEW

File: docs/configuration/configuration.md [hunk 1]
```diff
@@ -2839,3 +2843,11 @@ tls_config:
 There is a list of
 [integrations](https://prometheus.io/docs/operating/integrations/#remote-endpoints-and-storage)
 with this feature.
+
+### `<exemplars>`
+
+Note that exemplar storage is still considered experimental and must be enabled via `--enable-feature=exemplar-storage`.
+
+```yaml
```
Comment (LeviHarrison) [inline docs/configuration/configuration.md:2851]: There should be a closing <code>```</code> after the following documentation.

File: docs/configuration/configuration.md [hunk 2]
```diff
@@ -2839,3 +2843,11 @@ tls_config:
...
+  # Configures the maximum size of the circular buffer used to store exemplars for all series. Resizable during runtime.
+  [ max_exemplars: <int> | default = 100000 ]
```
Comment (LeviHarrison) [inline docs/configuration/configuration.md:2853]: No indent is needed here.

```suggestion
# Configures the maximum size of the circular buffer used to store exemplars for all series. Resizable during runtime.
[ max_exemplars: <int> | default = 100000 ]
```

**Review by LeviHarrison — APPROVED** [review]
Thanks!

**Comment by juliusv** [comment]
:+1: Thanks!

**Pull request merged**